### DISCUSSION TITLE
Does cluster mode support tls ？

### DISCUSSION THREAD

**Discussion by joyanhui** [apache/kvrocks#1897] [Q&A] [answered]
I was wondering if you could give me some information 

There is a configuration 'tls cluster yes' in Redis, which does not appear to have a similar configuration in kvrocks. I don't know if the connection between nodes is through TLS

Redis requires three ports,  tcp port  tls por  and cluster-announce-bus-port,does kvrocks have the same bus port

**Comment by PragmaTwice** [comment, marked as answer]
I think firstly I can give a general introduction of cluster mode in Kvrocks.

In Kvrocks, we have almost the same slot design (key distribution model) and cluster mode attributes as Redis, but we don't have a so-called *cluster bus* like Redis.

So there is almost no connection between cluster nodes. Hence the `cluster-announce-bus-port` is nothing in Kvrocks.

Currently we support TLS for server-client connections and replication connections. And it's expected to be enough for most situation in cluster mode. But TBH we currently have not supported TLS for slot migration yet. (And I'm not sure if I have some free time in the future to drive it.)

Please refer to https://kvrocks.apache.org/docs/cluster for more details.
For how to manage cluster nodes, refer to https://github.com/apache/kvrocks-controller.

**Comment by joyanhui** [comment]
hank you for your help
